dotcom-rendering/src/server/render.front.web.tsx

```diff
@@ -147,6 +147,11 @@ export const renderFront = ({
 
 	const keywords = front.config.keywords;
 
+	const canonicalUrl =
+		front.isNetworkFront && !!front.canonicalUrl
```

Comment:
Just a note to say that [`URL.canParse`](https://developer.mozilla.org/en-US/docs/Web/API/URL/canParse_static) may be safer than the simple existence of a string, as `new URL()` would throw an error for an invalid URL.

```suggestion
		front.isNetworkFront && URL.canParse(front.canonicalUrl ?? "")
```